DuckDuckGo/Application/AppDelegate.swift

```diff
@@ -90,6 +90,11 @@ final class AppDelegate: NSObject, NSApplicationDelegate {
     var privacyDashboardWindow: NSWindow?
 
     let activeRemoteMessageModel: ActiveRemoteMessageModel
+    let homePageSettingsModel = HomePage.Models.SettingsModel {
+        Task.detached { @MainActor in
```

Comment:
Rather than passing the action I would rather inject the WindowControllersManager (through a protocol that implements the required methods) in alternative. This would just be my preference though.

Reply:
After struggling with concurrency warnings I managed to do it finally :) thanks for pointing it out!